src/core/jpg.js

```diff
@@ -363,6 +363,12 @@ var JpegImage = (function jpegImage() {
       // find marker
       bitsCount = 0;
       marker = (data[offset] << 8) | data[offset + 1];
+      // Some bad images seem to pad Scan blocks with zero bytes, skip past
+      // those to attempt to find a valid marker (fixes issue4090.pdf).
+      while (marker === 0xFF && offset < data.length - 1) {
```

Comment:
I have a feeling that `marker === 0xFF` is not valid comparison here. It looks like the loop only skips one zero byte. Shall we make this an `if`?